### PR TITLE
update dependencies to latest substrate

### PR DESCRIPTION
**Pull request by insipx**

### CODE REVIEW

File: substrate-archive-backend/src/read_only_backend/blockchain_backend.rs [hunk 1]
```diff
@@ -181,6 +181,7 @@ impl<Block: BlockT, D: ReadOnlyDb> HeaderBackend<Block> for ReadOnlyBackend<Bloc
 			genesis_hash: meta.genesis_hash,
 			finalized_hash: meta.finalized_hash,
 			finalized_number: meta.finalized_number,
+			finalized_state: meta.finalized_state.clone(),
```
Comment (niklasad1) [inline substrate-archive-backend/src/read_only_backend/blockchain_backend.rs:184]:
```suggestion
			finalized_state: meta.finalized_state
```

**Review by niklasad1 — APPROVED** [review]

File: substrate-archive-backend/src/read_only_backend/blockchain_backend.rs [hunk 2]
```diff
@@ -181,6 +181,7 @@ impl<Block: BlockT, D: ReadOnlyDb> HeaderBackend<Block> for ReadOnlyBackend<Bloc
 			genesis_hash: meta.genesis_hash,
 			finalized_hash: meta.finalized_hash,
 			finalized_number: meta.finalized_number,
+			finalized_state: meta.finalized_state
```
Comment (niklasad1) [inline substrate-archive-backend/src/read_only_backend/blockchain_backend.rs:184]:
```suggestion
			finalized_state: meta.finalized_state,
```

**Review by TarikGul — APPROVED** [review]
lgtm

**Comment by insipx** [comment]
going to hold off on merging this for now, i'd like to fix the unit test that's failing

**Pull request merged**